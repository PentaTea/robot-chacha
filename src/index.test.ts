import { MockedApp } from 'koishi-test-utils'

// 创建一个无服务端的 App 实例
const app = new MockedApp()

// 创建一个 QQ 号为 123 的私聊会话
const session = app.createSession('user', 123)

// 还是刚刚那个例子
app.middleware(({ message, $send }, next) => {
    if (message === '天王盖地虎') return $send('宝塔镇河妖')
    return next()
})

test('example', async () => {
    // 将 foo 发送给机器人将会获得 bar 的回复
    await session.shouldHaveReply('天王盖地虎', '宝塔镇河妖')

    // 将 foo 发送给机器人将会获得某些回复
    await session.shouldHaveReply('天王盖地虎')

    // 将 foo 发送给机器人后将会获得与快照一致的回复
    await session.shouldMatchSnapshot('天王盖地虎')

    // 将 foo 发送给机器人将不会获得任何回复
    await session.shouldHaveNoReply('宫廷玉液酒')
})