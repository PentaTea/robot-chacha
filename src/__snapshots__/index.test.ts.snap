// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`example: 天王盖地虎 1`] = `
Array [
  "宝塔镇河妖",
]
`;
